src/services/transactions/hooks/useTrackTransactionStatus.ts

```diff
@@ -19,12 +25,27 @@ export interface UseTrackTransactionStatusArgsType {
 }
 
 export function useTrackTransactionStatus({
-  transactionId,
+  transactionId: txId,
   onSuccess,
   onFail,
   onCancelled,
   onTimedOut
 }: UseTrackTransactionStatusArgsType) {
+  const { signedTransactionsArray } = useGetSignedTransactions();
+  const isWalletProvider = getIsProviderEqualTo(LoginMethodsEnum.wallet);
+
+  const [sessionId] =
+    signedTransactionsArray.length > 0
+      ? signedTransactionsArray[signedTransactionsArray.length - 1]
+      : [];
+
+  /**
+   * Web wallet restores sessionId
+   */
+  const walletSessionId = txId ?? sessionId ?? null;
+
+  const transactionId = isWalletProvider ? walletSessionId : txId;
```

Comment:
I think we should skip this check, if txId is null, which will be the case with wallet, it will fallback to sessionId. If not, as it is for other providers, it will not trigger the ??. 

This can even be helpful, as someone who sends a tx and forgets to track the tx will get info about the tx anyway